src/devices/stepper.rs

```diff
@@ -0,0 +1,526 @@
+        })?;
+        Ok(())
+    }
+    /// Get current limit
+    pub fn get_current_limit(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getCurrentLimit(self.chan, &mut value) })?;
+        Ok(value)
+    }
+    /// Get minimum current limit
+    pub fn get_min_current_limit(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_getMinCurrentLimit(self.chan, &mut value)
+        })?;
+        Ok(value)
+    }
+    /// Get maximum current limit
+    pub fn get_max_current_limit(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_getMaxCurrentLimit(self.chan, &mut value)
+        })?;
+        Ok(value)
+    }
+
+    /// Set data interval
+    pub fn set_data_interval(&self, data_interval: u32) -> Result<()> {
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_setDataInterval(self.chan, data_interval)
+        })?;
+        Ok(())
+    }
+    /// Get data interval
+    pub fn get_data_interval(&self) -> Result<u32> {
+        let mut value = 0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getDataInterval(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Get minimum data interval
+    pub fn get_min_data_interval(&self) -> Result<u32> {
+        let mut value = 0;
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_getMinDataInterval(self.chan, &mut value)
+        })?;
+        Ok(value)
+    }
+
+    /// Get maximum data interval
+    pub fn get_max_data_interval(&self) -> Result<u32> {
+        let mut value = 0;
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_getMaxDataInterval(self.chan, &mut value)
+        })?;
+        Ok(value)
+    }
+
+    /// Set data rate
+    pub fn set_data_rate(&self, data_rate: f64) -> Result<()> {
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_setDataRate(self.chan, data_rate) })?;
+        Ok(())
+    }
+    /// Get data rate
+    pub fn get_data_rate(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getDataRate(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Get minimum data rate
+    pub fn get_min_data_rate(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getMinDataRate(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Get maximum data rate
+    pub fn get_max_data_rate(&self) -> Result<f64> {
+        let mut value = 0.0;
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_getMaxDataRate(self.chan, &mut value) })?;
+        Ok(value)
+    }
+
+    /// Set engaged
+    pub fn set_engaged(&self, engaged: bool) -> Result<()> {
+        let _v = match engaged {
+            true => 1,
+            false => 0,
+        };
```

Comment:
Nit: A `bool` can just be cast to an integer. It will always convert to 0 or 1.  So just `let v = engaged as i32;`.